### PR TITLE
fix freebsd build

### PR DESCRIPTION
**Pull request by jkartseva**
https://github.com/ClickHouse/ClickHouse/actions/runs/7256640256/job/19769624432?pr=58015#step:8:5466
```
Dec 19 03:09:41 /build/src/IO/AIO.cpp:140:15: error: use of undeclared identifier 'ErrnoException'; did you mean 'DB::ErrnoException'?
Dec 19 03:09:41   140 |         throw ErrnoException(DB::ErrorCodes::CANNOT_IOSETUP, "io_setup failed");
```

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-clickhouse-ci-1** [comment]
*This is an automated comment for commit 31f04b66c5cb468ef9fb532c5101e771e8b75f6b with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/58019/31f04b66c5cb468ef9fb532c5101e771e8b75f6b/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/7257121931">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/58019/31f04b66c5cb468ef9fb532c5101e771e8b75f6b/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

**Review by Algunenano — APPROVED** [review]

File: src/Common/ArrayCache.h [hunk 1]
```diff
@@ -179,13 +179,22 @@ class ArrayCache : private boost::noncopyable
...
-                throw ErrnoException(DB::ErrorCodes::CANNOT_MUNMAP, "Allocator: Cannot munmap {}", ReadableSize(size));
+            {
+                try
+                {
+                    throw DB::ErrnoException(DB::ErrorCodes::CANNOT_MUNMAP, "Allocator: Cannot munmap {}", ReadableSize(size));
+                }
+                catch (DB::ErrnoException &)
+                {
+                    DB::tryLogCurrentException(__PRETTY_FUNCTION__);
+                }
```
Comment (antonio2368) [inline src/Common/ArrayCache.h:196]: Why are you throwing a function just to catch it?

Comment (Algunenano) [reply src/Common/ArrayCache.h:196]: For logging. If we didn't catch it the most likely result will be program termination (as the caller won't catch it and directly).

Comment (antonio2368) [reply src/Common/ArrayCache.h:196]: Ah you want to log the stacktrace?
Okay, I assume there isn't any other way to do it.

Comment (Algunenano) [reply src/Common/ArrayCache.h:196]: Well, throwing the exception also adds an error to system.errors (and it's a real / important error), so throwing is a nice way to handle everything.

**Pull request merged**